internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -234,9 +234,23 @@ SegmentSealedImpl::LoadDeletedRecord(const LoadDeletedRecordInfo& info) {
     auto timestamps = reinterpret_cast<const Timestamp*>(info.timestamps);
 
     // step 2: fill pks and timestamps
+    ssize_t n = deleted_record_.ack_responder_.GetAck();
```

Comment:
The original delete implementation always happen in order.

With the new delete strategy, load segment and delete happens concurrently.

Will this still work?

Reply:
We tries to keep this behavior in this implementation. I'll double check that.

Reply:
> The original delete implementation always happen in order.
> 
> With the new delete strategy, load segment and delete happens concurrently.
> 
> Will this still work?

We still make sure deletions happen in order, but they can be overlapped, the order can be:
1. t1, t2
2. t2, t3
3. t1, t2, t3

can not be:
1. t1
2. t3
3. t2 t3